staging/src/k8s.io/api/events/v1/types.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Event is a report of an event somewhere in the cluster. It generally denotes some state change in the system.
```

Comment:
global doc comments:

1. start each field's godoc with the json-serialized name
2. note any validation requirements (max length, format restrictions, enum valid values, etc)
3. note the purpose of string fields when needed (are they machine-readable? human-readable?)

Reply:
I passed that, because it is exact copy of v1beta1 (which apparently passed API review in the past?)
So if we will be changing it here, we should also change the v1beta1 types.

Reply:
then we should also change the v1beta1 types... by promoting this to v1 we are effectively saying it is done... the documentation quality and completeness should reflect that.

Reply:
Hi Jordan, I'm a bit confused about the requirements you mentioned:
1. What do you mean by starting godoc with json-serialized name? Do you mean the doc for each field should start with the field name? Could you give me an example?
2. I didn't find any explicit docs talking about the validation requirements. Do you have any links or examples?
3. Do you mean I should mention in the godoc whether it's machine or human readable for string fields?

Reply:
In general, event producers and consumers should be able to write against the API by reading our API docs. As a concrete example, running this command should describe what I need to know to write against the `events.k8s.io/v1` API:

```sh
kubectl explain events --api-version events.k8s.io/v1
```

> What do you mean by starting godoc with json-serialized name? Do you mean the doc for each field should start with the field name? Could you give me an example?

For example, for the `ReportingController` field, since its JSON tag is `reportingController`, the doc would start with something like `reportingController contains the name of the controller ...`

> I didn't find any explicit docs talking about the validation requirements. Do you have any links or examples?

The format requirements, maximum lengths, or allowed values expressed in `ValidateEvent()` should be briefly described in the godocs, so a producer doesn't have to experiment with the API or read the source code to write against it successfully.

> Do you mean I should mention in the godoc whether it's machine or human readable for string fields?

Yes, especially for fields like `action` and `reason`, it's not clear if these are supposed to be selected from a known set, or are programmatic identifiers, etc

Reply:
Thanks Jordan for the detailed explanation! I've updated docs for both v1 and v1beta. Please take a look and let me know if there's anything I should change. Thanks!